business_requirement_deliverable/models/business.py

```diff
@@ -49,6 +50,46 @@ class BusinessRequirementResource(models.Model):
         string='Business Requirement Deliverable',
         ondelete='cascade'
     )
+    business_requirement_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Business Requirement',
+        required=True,
+    )
+    business_requirement_partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        related='business_requirement_id.partner_id',
+        string='Business Requirement',
+        store=True
+    )
+    business_requirement_project_id = fields.Many2one(
+        comodel_name='project.project',
+        related='business_requirement_id.project_id',
+        string='Business Requirement',
+        store=True
+    )
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        '''
+        Used the active_id 'business_requirement' field:
+        We get the BR current status and set Access rights.
+        make Business Requirement Resource Read Only if
+        BR is not in draft / confirmed state.
+        '''
+        result = super(BusinessRequirementResource, self).\
+            fields_view_get(view_id, view_type, toolbar=toolbar,
+                            submenu=submenu)
+        if view_type in ['form', 'tree']:
+            business_requirement = self.env['business.requirement'].\
+                browse(self._context.get('active_id')).state
+            if business_requirement not in ['draft', 'confirmed']:
+                doc = etree.XML(result['arch'])
+                nodes = doc.xpath("//tree")
+                for node in nodes:
+                    node.set('edit', '0')
+                result['arch'] = etree.tostring(doc)
+        return result
```

Comment:
I'm not in favour of this type of hacking.
It would be both simpler and customization-friendly to enforce this using Record Rules based on State.